src/main/cpp/src/get_json_object.cu

```diff
@@ -696,7 +400,7 @@ __device__ bool evaluate_path(json_parser& p,
   // There is a same constant in JSONUtil.java, keep them consistent when changing
   // Note: Spark-Rapids should guarantee the path depth is less or equal to this limit,
   // or GPU reports cudaErrorIllegalAddress
-  constexpr int max_path_depth = 16;
+  constexpr int max_path_depth = 8;
```

Comment:
Can we please split up these two changes so that we can evaluate the performance of each of them separately? I would really like to see max_path_depth be turned into a template parameter for the kernel. That way we can have a few different kernels and select the right one depending on the path done. That or we can do like @nvdbaranec did on the original version and set the amount of shared memory that would be needed as part of launching the kernel.  But that can be follow on work.

Reply:
> Can we please split up these two changes so that we can evaluate the performance of each of them separately?

Now, we combine [Key, Named] 2 paths to a single [Named] path, so the change from 16 to 8 is not a tuning.
And: 
[Subscribe, Index] => [Index]
[Subscribe, Wildcard] => [Wildcard]
No need to separate to two PRs.

> That or we can do like @nvdbaranec did on the original version and set the amount of shared memory that would be needed as part of launching the kernel. But that can be follow on work.

OK, will add in a follow-up.